src/Microsoft.OData.Core/UriParser/SemanticAst/ODataPath.cs

```diff
@@ -94,9 +107,18 @@ internal IList<ODataPathSegment> Segments
         /// Get the segments enumerator
         /// </summary>
         /// <returns>The segments enumerator</returns>
-        public IEnumerator<ODataPathSegment> GetEnumerator()
+        public Enumerator GetEnumerator()

```

Comment:
Another option is to simply expose `List<T>.Enumerator` but is that exposing implementation details?

I was going to say we have some default enumerators for wrapping standard data structures with struct-based enumerators:
```
public struct ODataListEnumerator<T> : IEnumerator<T>
```
```
public struct ODataArrayEnumerator<T> : IEnumerator<T>
```
But that would likewise expose implementation details... perhaps I'm over thinking, and as a result over complicating, this.

Reply:
I'm reverting my enumeration change - the indexer gets me what I need perf wise

Reply:
> is that exposing implementation details?

Yeah, while reviewing I kept trying to figure out how to do this. I ended up with like 3 nested generics and still wasn't any closer to a solution.